sympy/series/order.py

```diff
@@ -256,7 +256,12 @@ def __new__(cls, expr, *args, **kwargs):
                                 expr = Mul(*[a.expr for a in orders])
                             elif expr.is_Pow:
                                 expr = orders[0].expr**orders[1].expr
-                    expr = expr.as_independent(*args, as_Add=False)[1]
+
+                    # XXX: There is probably a better place to check for this:
+                    if expr.is_zero:
+                        expr = S.Zero
+                    else:
+                        expr = expr.as_independent(*args, as_Add=False)[1]
```

Comment:
```suggestion
                    c, expr = expr.as_independent(*args, as_Add=False)
                    if c.is_zero:
                        expr = S.Zero
```

Reply:
What effect would this suggested change have?

Reply:
Note that these lines were already changed to something else on master. Let's see if there is still any need to change anything here.